options/options.c

```diff
@@ -715,6 +715,10 @@ const m_option_t mp_opts[] = {
                       0, INT_MAX, ({"auto", -1})),
 #endif
 
+#if HAVE_VAAPI_HWACCEL
+    OPT_STRING("vaapi-device", vaapi_device, M_OPT_FILE),
```

Comment:
I should clarify, I'm not saying you need to implement all the other APIs, just set up the infrastructure so that it's straightforward for others to do so.